Tests/DatadogTests/Datadog/RUM/RUMMonitor/Scopes/RUMUserActionScopeTests.swift

```diff
@@ -577,4 +579,37 @@ class RUMUserActionScopeTests: XCTestCase {
         XCTAssertNil(writer.events(ofType: RUMActionEvent.self).first)
         XCTAssertFalse(callbackCalled)
     }
+
+    // MARK: - Actions with Frustrations
+
+    func testGivenTapUserActionWithError_itWritesErrorTapFrustration() throws {
```

Comment:
I think it might make sense to have 2 distinct tests:
* _**when** error occurs within the lifespan of an action, **then** it adds error tap frustration_
* _**when** error occurs later than the lifespan of an action, **then** it DOES NOT add error tap frustration_

This way we will assert both new behaviours explicitly and exclusively, instead of asserting "not" as a side-effect in other tests. Otherwise, IMHO the convention proposed in this PR isn't well sustainable - why to assert "not" in the 2 tests proposed, but not in other? There will be always more than one tests to assert "not".

Reply:
Indeed, I've improve the coverage here. I'm also adding this one
- when error occurs on a non discrete action, then it DOES NOT add error tap frustration